src/axom/core/tests/utils_locale.hpp

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+//-----------------------------------------------------------------------------
+//
+// file: utils_locale.hpp
+// Checks locales and prints out related info
+//
+//-----------------------------------------------------------------------------
+
+#include "axom/core/utilities/StringUtilities.hpp"
+#include "axom/core/utilities/System.hpp"
+
+#include "axom/fmt.hpp"
+
+#include <locale>
+#include <string>
+#include <cstdio>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <array>
+#include <vector>
+#include <set>
+#include <algorithm>
+
+#include "gtest/gtest.h"
+
+#ifdef _WIN32
+  #include <Windows.h>
+#endif
+
+namespace
+{
+#ifdef __linux__
+/**
+ * \brief Utility function to capture output of system call
+ * \note Adapted from https://stackoverflow.com/a/478960
+ */
+std::string execute_command(const std::string& cmd)
+{
+  std::array<char, 128> buffer;
+  std::string result;
+
+  std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd.c_str(), "r"), pclose);
+  if(!pipe)
+  {
+    throw std::runtime_error(
+      axom::fmt::format("popen() failed for command '{}'", cmd));
+  }
+
+  while(fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr)
+  {
+    result += buffer.data();
+  }
+
+  return result;
+}
+#endif  // __linux__
+
+#ifdef _WIN32
+/// callback function for EnumSystemLocaleEx call in enumerate_locales_windows test
+BOOL CALLBACK MyFuncLocaleEx(LPWSTR pStr, DWORD dwFlags, LPARAM strvec_ptr)
+{
+  using StrVec = std::vector<std::wstring>;
+  reinterpret_cast<StrVec*>(strvec_ptr)->push_back(pStr);
```

Comment:
Why can't we do static_cast here instead?

Reply:
Thanks -- I'll give it a try.

Reply:
Reinterpret_cast should be avoided if possible. We even have that in our coding guidelines...https://axom.readthedocs.io/en/develop/docs/sphinx/coding_guide/sec08_design_implement.html#avoid-c-style-casts-const-cast-and-reinterpret-cast  😄

Reply:
Unfortunately, static_cast didn't work since the types are unrelated:
>error C2440: 'static_cast': cannot convert from 'LPARAM' to 'StrVec *'
Conversion from integral type to pointer type requires reinterpret_cast, C-style cast or parenthesized function-style cast

>error C2440: 'static_cast': cannot convert from 'std::vector<std::wstring,std::allocator<std::wstring>> *' to 'LPARAM'
There is no context in which this conversion is possible

See also: https://stackoverflow.com/questions/12147108/passing-this-pointer-as-lparam

I guess I'll try a c-style cast (?)
(Our docs seem to imply the c-style cast is worse than `reinterpret-cast`, since the latter says "...unless absolutely necessary")

Which would you prefer?